packages/binding-http/src/http-server.ts

```diff
@@ -311,9 +312,9 @@ export default class HttpServer implements ProtocolServer {
                     ];
                 }
                 if (!thing.forms) {
-                    thing.forms = [];
+                    thing.forms = new Array() as [FormElementRoot];
                 }
-                thing.forms.push(form);
+                thing.forms.push(form as FormElementRoot);
```

Comment:
why is TD.Form not compatible with FormElementRoot already ?

Reply:
This is a good question and and somewhat also complicated given that we re-use the types from [TD JSON Schema](https://github.com/w3c/wot-thing-description/blob/main/validation/td-json-schema-validation.json).

There we have  *different* forms for root, properties, actions and events.

https://github.com/w3c/wot-thing-description/blob/0600e4384c129bf1dc1fa4fa1a34074d6b684afa/validation/td-json-schema-validation.json#L254

https://github.com/w3c/wot-thing-description/blob/0600e4384c129bf1dc1fa4fa1a34074d6b684afa/validation/td-json-schema-validation.json#L437

..

The question is whether we should have *different* `form`s also.

Reply:
see https://github.com/eclipse/thingweb.node-wot/issues/561